src/Substance.hs

```diff
@@ -4,535 +4,450 @@
 
 {-# OPTIONS_HADDOCK prune #-}
 module Substance where
--- module Main (main) where -- for debugging purposes
+--module Main (main) where -- for debugging purposes
 -- TODO split this up + do selective export
 
 import Utils
 import System.Process
+import Data.Void
 import Control.Monad (void)
 import System.IO -- read/write to file
 import System.Environment
 import Control.Arrow ((>>>))
 import System.Random
 import Debug.Trace
 import Data.List
-import Data.Maybe (fromMaybe)
+import Data.Maybe
 import Data.Typeable
 import Text.Megaparsec
 import Text.Megaparsec.Char
 import Text.Megaparsec.Expr
+import Env
 -- import Text.PrettyPrint
-import Text.PrettyPrint.HughesPJClass hiding (colon, comma, parens, braces)
+--import Text.PrettyPrint.HughesPJClass hiding (colon, comma, parens, braces)
+import qualified Dsll as D
 import qualified Data.Map.Strict as M
 import qualified Text.Megaparsec.Char.Lexer as L
 
---------------------------------------------------------------------------------
--- Substance AST
+--------------------------------------- Substance AST ---------------------------------------
 
--- | A Substance program is a list of statements
+data ValConstructorName = ValConst String             -- “Cons”, “Times”
```

Comment:
I am confused about the relationship b/t Sub and SubCore. Which one is used? Which one is deprecated? Seems like they should be merged